samples/smp/pi/sample.yaml

```diff
@@ -0,0 +1,18 @@
+sample:
+  description: Calculation of Pi independently in
+    a number of threads
+  name: SMP Pi
+common:
+    tags: introduction
+    harness: console
+    harness_config:
+      type: multi_line
+      ordered: yes
+      regex:
+        - "Calculate first [0-9]+ digits of Pi independently by [0-9]+ threads.(.*)"
+        - "Pi value calculated by thread #[0-9]+: [0-9]+(.*)"
+        - "All [0-9]+ threads executed by [0-9]+ cores in [0-9]+ msec(.*)"
+tests:
+  sample.smp_pi:
+    tags: introduction
+    platform_whitelist: nsim_hs_smp qemu_x86_64
```

Comment:
instead of whitelisting, this should be 

`filter: CONFIG_SMP and CONFIG_MP_NUM_CPUS > 1`

which will allow new smp configuration to run without having to change this file again.

Reply:
Well I whitelisted those 2 platforms intentionally because:
1. Not all boards have enough RAM and here we need quite a lot due to large tables used for calculations. If I'm not mistaken we're talking about more than 100 KiB, for example built for `qemu_x86_64` it is:
   ```
    # size zephyr/zephyr.elf
      text    data     bss     dec     hex filename
     37977     552  133808  172337   2a131 zephyr/zephyr.elf
   ```

2. Some boards couldn't be used in generic sanity checking like ARC HSDK because most of people doesn't own them. But in that sense `nsim_hs_smp` should be removed from whilte-list as well I guess... but funny enough if `sanitycheck` is executed w/o `--all` then only the first whitelisted platform is used. Not sure if that's an expected behavior though...